.github/workflows/cpp.yml

```diff
@@ -117,6 +117,10 @@ jobs:
         run: |
           cd cpp/examples/minimal_build
           docker-compose run --rm minimal
+      - name: Run minimal example (static, system dependencies)
+        run: |
+          cd cpp/examples/minimal_build
+          docker-compose run --rm static-system-dependency
```

Comment:
Does this reproduce this problem?
I couldn't reproduce this problem by this without your `cpp/src/arrow/CMakeLists.txt` change.

Reply:
I admit I can't test this particular part locally.
FTR: is this already run elsewhere in the CI configurations?

Reply:
Yes:

* https://github.com/apache/arrow/pull/13707#issuecomment-1196237298
* https://github.com/ursacomputing/crossbow/runs/7531519590?check_suite_focus=true